StRoot/StEvent/StEnumerations.cxx

```diff
@@ -30,23 +84,5 @@ StDetectorId detectorIdByName(const char *name)
 //_____________________________________________________________________________
 void detectorIdInit()
 {
-const char *paths[] = {
-                  "./detectorId.C",
-    "./StRoot/macros/detectorId.C",
-"$STAR/StRoot/macros/detectorId.C",
-                                0};
-
-  ids[0] = -1;
-  for (int ifa=0;paths[ifa];ifa++) {
-    TString path(paths[ifa]);
-    gSystem->ExpandPathName(path);
-    int notExi = gSystem->AccessPathName(path.Data());
-    if (notExi) continue;
-    ids[0]=0;
-    TString com;
-    com.Form("%s((int*)%p,(const char**)%p)",path.Data(),ids,cds);
-    gROOT->Macro(com.Data());
-    assert(ids[0]);
```

Comment:
That different programming languages evaluate negative numbers as true or false differently always drives me to be clear if there's the possibility of a negative value, i.e. `asserts(ids[0]!=0);`, but it's just a suggestion because C/C++ behaves consistently.